test/jdk/sun/nio/ch/ChannelInputStream.java

```diff
@@ -0,0 +1,143 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8268435
+ * @summary Verify ChannelInputStream methods
+ * @library ..
+ * @library /test/lib
+ * @build jdk.test.lib.RandomFactory
+ * @modules java.base/jdk.internal.util
+ * @run testng/othervm -Xmx6G ChannelInputStream
+ * @key randomness
+ */
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Random;
+import jdk.internal.util.ArraysSupport;
+
+import jdk.test.lib.RandomFactory;
+
+import org.testng.Assert;
+import static org.testng.Assert.*;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class ChannelInputStream {
+
+    static final Random RAND = RandomFactory.getRandom();
+
+    static File createFile(long size) throws IOException {
+        File file = File.createTempFile("foo", ".bar");
+        file.deleteOnExit();
+        RandomAccessFile raf = new RandomAccessFile(file, "rw");
+        raf.setLength(size);
+        raf.close();
+        return file;
+    }
+
+    @DataProvider
+    public Object[][] provider() throws IOException {
+
+        Object[][] result = new Object[][] {
+            {createFile(0L)},
+            {createFile(RAND.nextInt(Short.MAX_VALUE))},
+            {createFile(ArraysSupport.SOFT_MAX_ARRAY_LENGTH)},
+            {createFile((long)Integer.MAX_VALUE + 27)}
+        };
+
+        return result;
+    }
+
+    @Test(dataProvider = "provider")
+    public void readAllBytes(File file) throws IOException {
+        try (FileInputStream fis = new FileInputStream(file);
+             FileChannel fc = fis.getChannel();
+             InputStream cis = Channels.newInputStream(fc)) {
+            long size = Files.size(file.toPath());
+            if (size == 0L) {
+                byte[] bytes = cis.readAllBytes();
+                assertNotNull(bytes);
+                assertEquals(bytes.length, 0);
+            } else if (size > Integer.MAX_VALUE) {
+                expectThrows(OutOfMemoryError.class, () -> cis.readAllBytes());
+            } else {
+                byte[] fisBytes = fis.readAllBytes();
+                fc.position(0L);
+                byte[] cisBytes = cis.readAllBytes();
+                assertTrue(Arrays.equals(fisBytes, cisBytes));
+
+                // check behavior at what should be EOF
+                cisBytes = cis.readAllBytes();
+                assertNotNull(cisBytes);
+                assertEquals(cisBytes.length, 0);
+            }
+        } finally {
+            file.delete();
+        }
```

Comment:
It might be easy to maintain if we split this up so there are separate tests for empty files, OOME, etc.
Are you planning to update the test cover the case that the initial position is > 0?
Has the setup been copied from an older test, I'm wondering why it needs to use File and FileChannel::getChannel as you should be able to use FileChannel.open.